### ISSUE TITLE
[BUG] Security plugin transport layer(multi-node cluster) does not work due to missing required reflective access (JDK17)

### ISSUE THREAD

**Issue by cliu123** [opensearch-project/security#1653]
**Describe the bug**
Security plugin 1.3.0.0 does not work on OpenSearch 1.3.0 multi-node clusters without required reflective access.
This issue was originated from: https://github.com/opensearch-project/security/issues/1619

**To Reproduce**
1. Create a OpenSearch 1.3.0 cluster with at least 2 nodes.
2. Install security plugin 1.3.0.0(security plugin repo [main branch](https://github.com/opensearch-project/security)) on each of the nodes.
3. Send a basic request(such as `_cat/nodes` or `_cat/indices`) to the OpenSearch cluster.
4. See error:
```
% curl -k -XGET -u admin:admin https://localhost:9200/_cat/indices
{"error":{"root_cause":[{"type":"security_exception","reason":"Unexpected exception indices:monitor/stats"}],"type":"security_exception","reason":"Unexpected exception indices:monitor/stats"},"status":500}% 
```
Stacktrace in logs:
```
org.opensearch.OpenSearchSecurityException: Unexpected exception cluster:monitor/nodes/info
         at org.opensearch.security.filter.SecurityFilter.apply0(SecurityFilter.java:376) [opensearch-security-1.3.0.0.jar:1.3.0.0]
         at org.opensearch.security.filter.SecurityFilter.apply(SecurityFilter.java:154) [opensearch-security-1.3.0.0.jar:1.3.0.0]
         at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:192) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.performanceanalyzer.action.PerformanceAnalyzerActionFilter.apply(PerformanceAnalyzerActionFilter.java:99) [opensearch-performance-analyzer-1.3.0.0.jar:1.3.0.0]
         at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:192) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.support.TransportAction.execute(TransportAction.java:169) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.support.TransportAction.execute(TransportAction.java:97) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.client.node.NodeClient.executeLocally(NodeClient.java:108) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.client.node.NodeClient.doExecute(NodeClient.java:95) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:433) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.client.support.AbstractClient$ClusterAdmin.execute(AbstractClient.java:730) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.client.support.AbstractClient$ClusterAdmin.nodesInfo(AbstractClient.java:813) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.rest.action.cat.RestNodesAction$1.processResponse(RestNodesAction.java:127) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.rest.action.cat.RestNodesAction$1.processResponse(RestNodesAction.java:115) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.rest.action.RestActionListener.onResponse(RestActionListener.java:60) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:103) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:97) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.ActionListener$2.onResponse(ActionListener.java:104) [opensearch-1.3.0.jar:1.3.0]
         at org.opensearch.action.ActionListener.completeWith(ActionListener.java:351) [opensearch-1.3.0.jar:1.3.0]
 
```
Digging into the reason of the error:
1. 
```
{
  "error": {
    "root_cause": [
      {
        "type": "security_exception",
        "reason": "Unexpected exception cluster:monitor/nodes/info"
      }
    ],
    "type": "security_exception",
    "reason": "Unexpected exception cluster:monitor/nodes/info",
    "caused_by": {
      "type": "exception_in_initializer_error",
      "reason": null,
      "caused_by": {
        "type": "inaccessible_object_exception",
        "reason": "Unable to make field private java.lang.String java.io.ObjectStreamClass.name accessible: module java.base does not \"opens java.io\" to unnamed module @5b0575d0"
      }
    }
  },
  "status": 500
}

```
2. 
```
{
  "error": {
    "root_cause": [
      {
        "type": "security_exception",
        "reason": "Unexpected exception cluster:monitor/nodes/info"
      }
    ],
    "type": "security_exception",
    "reason": "Unexpected exception cluster:monitor/nodes/info",
    "caused_by": {
      "type": "no_class_def_found_error",
      "reason": "Could not initialize class org.opensearch.security.support.Base64Helper$DescriptorNameSetter"
    }
  },
  "status": 500
}
```
**Expected behavior**
Requests gets expected response like the following:
```
% curl -k -XGET -u admin:admin https://localhost:9200/_cat/indices                                                                                    
green open security-auditlog-2022.02.19 l9rrGwzOR9C4ZWqMyMYHwg 1 1  1 0  23.2kb  11.6kb
green open security-auditlog-2022.02.18 qvL83MJ1RbG1YgOW4MixAQ 1 1 22 0 244.9kb 154.3kb
green open .kibana_1                    peXycy4TSoWTYm8Uh6TemQ 1 1  1 0    10kb     5kb
green open .opendistro_security         3RaQH4lCQuayi6CKLXptWg 1 1  9 0 106.1kb    53kb
```

**Plugins**
Please list all plugins currently enabled.
Security plugin

**Host/Environment (please complete the following information):**
 - OS: [Linux]

**Additional context**
[PR#1278](https://github.com/opensearch-project/security/pull/1278) in security plugin requires these reflective access since the changes use reflection. But reverting the PR is not an option since the PR itself supports OpenSearch backward compatibility for ODFE. Without this PR(#1278), security plugin would not work in mixed cluster(ODFE nodes + OpenSearch nodes), which would fail rolling upgrade from ODFE to OpenSearch. 
In order to fix the issue, JVM Options where OpenSearch cluster runs need to be appended with something like [this](https://github.com/opensearch-project/security/commit/f8ec31e383f5354fd849c68fc949425c07fc7626) to grant the reflective access. Security plugin had exactly same error message in Integration Tests, which was resolved by such changes.

**Comment by reta** [comment]
@dblock gotcha, we could make `org.opensearch.security.support.Base64Helper` a good citizen here to conform to JDK-17 rules

**Comment by cliu123** [comment]
@dblock @reta Fixing this on security plugin would not resolve the root cause introduced by JDK 17 in runtime. Other plugins that require reflective accesses would still have this issue, for example [PA plugin](https://github.com/opensearch-project/performance-analyzer/issues/144). And reverting the related PR would break backward compatibility of security plugin [like I mentioned](https://github.com/opensearch-project/security/issues/1653#issuecomment-1055818563).
Cc: @davidlago @sruti1312 for any inputs.

**Comment by reta** [comment]
@cliu123 but we have to fix this access violation in any case, right? (in all the places)

**Comment by davidlago** [comment]
My vote would be to push JDK17 to 2.0 and open issues in all affected plugins like security, PA etc to fix the access violation by 2.0

**Comment by cliu123** [comment]
> @cliu123 but we have to fix this access violation in any case, right? (in all the places)

@reta Yes, the access violation can by fixed only by reverting the related PR in security plugin. And 2.0 will be a better timing to do this as @setiah [explained](https://github.com/opensearch-project/security/issues/1653#issuecomment-1055818575).
> Also, for 2.0, we do not need the #1278 as the mixed cluster scenario with ODFE and 2.0 version nodes would not arise (PS: Since to upgrade from ODFE, ODFE 1.13 -> OpenSearch 1.x -> OpenSearch2.0). 

If reverting the PR in 1.3.0 release, users would not be able to do rolling upgrade from ODFE to OpenSearch 1.3.0. If JDK 17 is postpone to 2.0, then security plugin would be able to revert the PR since the supported upgrade path is `ODFE 1.13 -> OpenSearch 1.x -> OpenSearch2.0`.

**Comment by setiah** [comment]
> @reta This is all part of [opensearch-project/opensearch-plugins#64](https://github.com/opensearch-project/opensearch-plugins/issues/64) in which we're trying to switch builds to 11 and upgrade runtime to 17. This issue prevents upgrading the runtime to 17. I'm moving this issue to security and let's figure out whether we should fix this or ship 1.3 with JDK 11 instead.

Thanks @reta for the quick response. Since it defaults to jdk17 runtime now (bundled jdk), it runs OpenSearch and plugins on it by default. Setting `RUNTIME_JAVA_HOME` to a different (lower) version explicitly might work, but a user might have to set that explicitly (and also have that jdk installed beforehand). IMO it still breaks the existing behavior for users in terms of usability (for a minor version release).

**Comment by reta** [comment]
@setiah many do not use bundled JDK, moving to 11 [1]  "solves" our problem but not user's problem, JDK-17 is the version many are moving to in production ... 

[1] https://github.com/opensearch-project/OpenSearch/pull/2301

**Comment by cliu123** [comment]
@dblock Security plugin has another tracking issue for this matter #1619. Would it make more sense to move this one to OpenSearch repo to track the fix? I don't have the permission to transfer this to OpenSearch repo though.

**Comment by cliu123** [comment]
@ylwu-amzn [The PR](https://github.com/opensearch-project/OpenSearch/pull/2301) has been merged. Has [the issue](https://github.com/opensearch-project/security/issues/1619) been resolved?

**Comment by ylwu-amzn** [comment]
We are using 1.2.4 doing PenTest. Please keep that issue open once you confirm the issue resolved on 1.3.

**Comment by cliu123** [comment]
> We are using 1.2.4 doing PenTest. Please keep that issue open until you confirm the issue resolved on 1.3.

That issue that you created has been closed since the root cause(JDK 17 in runtime) has been identified, and this issue was created to track the resolution of the root cause. Please close this issue whenever you confirm that your issue is resolved.

**Comment by ylwu-amzn** [comment]
@cliu123 , busy with finishing coding now, so will keep that issue open for a while. If you are sure that issue is gone, you can close that issue directly. If we found any issue, will reopen or create a new one. Thanks.

**Comment by dblock** [comment]
We've given up on JDK 17 for OpenSearch v.1.3.0 and [downgraded](https://github.com/opensearch-project/OpenSearch/pull/2301) OpenSearch 1.x branch to JDK 11. Re-labelling this issue as 2.0.

**Comment by cliu123** [comment]
@ylwu-amzn confirmed that the issue has been resovled. Closing the issue.

**Comment by cliu123** [comment]
The issue got closed because 1.3 got unblocked by rulling out JDK 17 support.
Re-opening this issue for 2.0. 
In order to support JDK 17 in 2.0, security plugin would have to revert a [PR](https://github.com/opensearch-project/security/pull/1278) from 2.0 that is needed to support rolling upgrade from ODFE to OpenSearch 1.x. With reverting this PR from 2.0, security plugin wouldn't support rolling upgrade support from ODFE to 2.0 directly, but the rolling upgrade from ODFE to OpenSearch 1.x would still be supported. Since we're moving towards 2.0, this can be one of the breaking changes. Please confirm this. @CEHENKLE @dblock
Cc: @davidlago

**Comment by cliu123** [comment]
Closing the issue with #1691
